example-app/src/main/java/com/adyen/checkout/example/ui/main/MainActivity.kt

```diff
@@ -85,21 +85,10 @@ class MainActivity : AppCompatActivity(), DropInCallback {
         return super.onOptionsItemSelected(item)
     }
 
-    override fun onDropInResult(dropInResult: DropInResult?) {
-        if (dropInResult == null) return
-        when (dropInResult) {
-            is DropInResult.CancelledByUser -> Toast.makeText(this, "Canceled by user", Toast.LENGTH_SHORT).show()
-            is DropInResult.Error -> Toast.makeText(this, dropInResult.reason, Toast.LENGTH_SHORT).show()
-            is DropInResult.Finished -> Toast.makeText(this, dropInResult.result, Toast.LENGTH_SHORT).show()
-        }
-    }
+    override fun onDropInResult(dropInResult: DropInResult?) = viewModel.onDropInResult(dropInResult)
 
     private fun onViewState(viewState: MainViewState) {
         when (viewState) {
-            is MainViewState.Error -> {
-                setLoading(false)
-                Toast.makeText(this, viewState.message, Toast.LENGTH_SHORT).show()
-            }
             MainViewState.Loading -> setLoading(true)
```

Comment:
did you double check for if loading not dismissed when any error happens? `setLoading(false)`